ray-operator/pkg/features/features.go

```diff
@@ -0,0 +1,48 @@
+package features
+
+import (
+	"testing"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+)
+
+const (
+	// owner: @rueian @kevin85421 @andrewsykim
+	// rep: https://github.com/ray-project/enhancements/pull/54
+	// alpha: v1.2
+	//
+	// Enables new conditions in RayCluster status
+	RayClusterStatusConditions featuregate.Feature = "RayClusterStatusConditions"
+)
+
+func init() {
+	runtime.Must(utilfeature.DefaultMutableFeatureGate.Add(defaultFeatureGates))
+}
+
+var defaultFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
+	RayClusterStatusConditions: {Default: false, PreRelease: featuregate.Alpha},
+}
+
+// SetFeatureGateDuringTest is a helper method to override feature gates in tests.
+func SetFeatureGateDuringTest(tb testing.TB, f featuregate.Feature, value bool) func() {
+	return featuregatetesting.SetFeatureGateDuringTest(tb, utilfeature.DefaultFeatureGate, f, value)
```

Comment:
It feels weird to import the `testing` and the `featuregatetesting` package outside a `_test.go` file. I believe that will have some impact on the compilation time. Is this a common practice while implementing a custom feature gate with the `utilfeature` package? Or could we just move them into a `_test.go` file?

Reply:
> I believe that will have some impact on the compilation time.

This is negligible impact to compilation time

> Is this a common practice while implementing a custom feature gate with the utilfeature package? Or could we just move the SetFeatureGateDuringTest into a _test.go file?

I believe it's common to do this because many test packages will execute code that checks the feature gate and will compile this package already. If you move it into `_test.go` files, then each test will have to carry a copy of this method or import it from a common test package.